packages/oauth/src/index.ts

```diff
@@ -312,30 +319,41 @@ export class InstallProvider {
     req: IncomingMessage,
     res: ServerResponse,
     options?: CallbackOptions,
+    installOptions?: InstallURLOptions,
   ): Promise<void> {
     let parsedUrl;
     let code: string;
     let flowError: string;
     let state: string;
-    let installOptions: InstallURLOptions;
-
     try {
       if (req.url !== undefined) {
-        parsedUrl = new URL(req.url);
+        parsedUrl = new URL(req.url, `https://${req.headers.host}`);
```

Comment:
@srajiang Can you move this to a private method like `const searchParams = extractSearchParams(req)` or have some comment saying that protocol/host in the `parsedUrl` may not be accurate here?